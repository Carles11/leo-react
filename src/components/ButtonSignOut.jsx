import React from 'react'
import PropTypes from 'prop-types'

const ButtonSignOut = (props) => {
  const { handleClick, label } = props
  const labelTxt = label ? label.toUpperCase() : ''

  return (
    <button
      className="btn-signOut"
      aria-label="Cerrar el Panel de administraciÃ³n"
      onClick={handleClick}
    >
      {labelTxt}
      <span className="icon-stand-by" />
    </button>
  )
}

ButtonSignOut.defaultProps = {
  label: '',
}

ButtonSignOut.propTypes = {
  handleClick: PropTypes.func.isRequired,
}

export default ButtonSignOut
