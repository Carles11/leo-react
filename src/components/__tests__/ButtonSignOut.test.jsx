import React from 'react'
import { shallow } from 'enzyme'
import ButtonSignOut from '../ButtonSignOut'

const properties = {
  handleClick: jest.fn(),
}

describe('<ButtonSignOut />', () => {
  it('should render itself', () => {
    const component = shallow(<ButtonSignOut {...properties} />)
    expect(component).toMatchSnapshot()
  })
})
