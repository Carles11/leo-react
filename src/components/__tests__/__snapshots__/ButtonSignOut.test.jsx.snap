// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSignOut /> should render itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonSignOut
    handleClick={[Function]}
    label=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Cerrar el Panel de administración",
      "children": Array [
        "",
        <span
          className="icon-stand-by"
/>,
      ],
      "className": "btn-signOut",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon-stand-by",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Cerrar el Panel de administración",
        "children": Array [
          "",
          <span
            className="icon-stand-by"
/>,
        ],
        "className": "btn-signOut",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon-stand-by",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
